packages/@lwc/ssr-compiler/src/compile-template/transformers/component.ts

```diff
@@ -72,26 +92,33 @@ function getChildAttrsOrProps(
     attrs: (IrAttribute | IrProperty)[],
     cxt: TransformerContext
 ): EsObjectExpression {
-    const objectAttrsOrProps = attrs.map((attr) => {
-        const key = isValidIdentifier(attr.name) ? b.identifier(attr.name) : b.literal(attr.name);
-        if (attr.value.type === 'Literal' && typeof attr.value.value === 'string') {
-            const value =
-                attr.name === 'style'
-                    ? normalizeStyleAttribute(attr.value.value)
-                    : attr.value.value;
-            return b.property('init', key, b.literal(value));
-        } else if (attr.value.type === 'Literal' && typeof attr.value.value === 'boolean') {
-            return b.property(
-                'init',
-                key,
-                b.literal(attr.type === 'Attribute' ? '' : attr.value.value)
-            );
-        } else if (attr.value.type === 'Identifier' || attr.value.type === 'MemberExpression') {
-            const propValue = expressionIrToEs(attr.value, cxt);
-            return b.property('init', key, propValue);
-        }
-        throw new Error(`Unimplemented child attr IR node type: ${attr.value.type}`);
-    });
+    const objectAttrsOrProps = attrs
+        .map(({ name, value, type }) => {
+            const key = isValidIdentifier(name) ? b.identifier(name) : b.literal(name);
+            if (value.type === 'Literal' && typeof value.value === 'string') {
+                let literalValue = value.value;
+                if (name === 'style') {
+                    literalValue = normalizeStyleAttributeValue(literalValue);
+                } else if (name === 'class') {
+                    literalValue = normalizeClassAttributeValue(literalValue);
+                    if (literalValue === '') {
+                        return; // do not render empty `class=""`
+                    }
+                }
+                return b.property('init', key, b.literal(literalValue));
+            } else if (value.type === 'Literal' && typeof value.value === 'boolean') {
+                if (name === 'class') {
+                    return; // do not render empty `class=""`
+                }
+
+                return b.property('init', key, b.literal(type === 'Attribute' ? '' : value.value));
+            } else if (value.type === 'Identifier' || value.type === 'MemberExpression') {
+                const propValue = expressionIrToEs(value, cxt);
+                return b.property('init', key, propValue);
+            }
+            throw new Error(`Unimplemented child attr IR node type: ${value.type}`);
+        })
+        .filter(Boolean) as EsProperty[];
```

Comment:
```suggestion
        .filter(<T>(v: T): v is NonNullable<T> => v)
```
I thiiink this works?

Reply:
I think this is actually less clear than the previous version. It's a shame TS doesn't understand `.filter(Boolean)`.

Reply:
```suggestion
        .filter(v => v is EsProperty => Boolean(v));
```

Reply:
Didn't work. I'm not going to fight TypeScript on this one